src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -154,6 +156,7 @@ void EnsureDirectoriesExists()
         }
 
         [DebuggerNonUserCode]
+        [SuppressMessage("Code", "PCR0010:Single CancellationToken parameters should be named cancellationToken", Justification = "Multiple cancellation tokens flying around.")]
```

Comment:
curious why the "or ending with `CancellationToken`" condition didn't make it into the analyzer. I'm concerned about the friction introduced by requiring a suppression to be more expressive, especially in a non-public API. This will be an issue in every single message pump.